src/backend/libc/net/msghdr.rs

```diff
@@ -36,8 +36,10 @@ pub(crate) fn with_recv_msghdr<R>(
     let res = f(&mut msghdr);
 
     // Reset the control length.
-    unsafe {
-        control.set_control_len(msghdr.msg_controllen.try_into().unwrap_or(usize::MAX));
+    if res.is_ok() {
+        unsafe {
+            control.set_control_len(msghdr.msg_controllen.try_into().unwrap_or(usize::MAX));
+        }
```

Comment:
I'm not sure if the contents of the control buffer are defined on error. I think it would be safest to set the buffer's length to zero on error.

Reply:
`set_control_len` just sets the `length` (and `read`) field of the `RecvAncillaryBuffer`, which isn't part of the buffer that libc will be touching so it won't be changed by the `recvmsg`  call if this `set_control_len` line isn't run.

And for a newly allocated `RecvAncillaryBuffer`, it will already be `0`. So there's no need to set it, in that case.

If the same `RecvAncillaryBuffer` is re-used in multiple calls, I'm not sure how that works currently. Should check that doesn't leak file descriptors or anything.

Reply:
We should probably be clearing out all of the inner data before we do anything with it.

Reply:
Added a commit that clears the `RecvAncillaryBuffer` at the start of the recv operation.